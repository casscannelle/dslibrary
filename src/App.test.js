import { render, screen, fireEvent } from '@testing-library/react';
import App from './App';

//Testes Simples
test("Botão com getByRole", () => {
const { getByRole } = render(<App />);
const button = getByRole("button", { name: "Enviar" });
expect(button).toBeInTheDocument();
expect(button).toHaveTextContent("Enviar");
});

test('Encontrar texto na página', () => {
  render(<App />);
  const linkElement = screen.queryByText(/Página/i);
  expect(linkElement).toBeInTheDocument();
});

test('Busca por Placeholder', () => {
  render(<App />);
  const linkElement = screen.getByPlaceholderText(/Nome/i);
  expect(linkElement).toBeInTheDocument();
});

//Teste do caminho Feliz
test('Renderiza botão quando clicado', () => {
  const { getByText, queryByText } = render(<App />);
  const textoAntesDoClique = queryByText("Botão Clicado");
  expect(textoAntesDoClique).toBeNull();
  
  const botao = getByText("Clique aqui");
  
  fireEvent.click(botao);

  const textoAposClique = getByText("Botão Clicado");
  expect(textoAposClique).toBeInTheDocument();
});


//Teste que retorna Erro
test("renders button correctly and does not show text when clicked", () => {
  const { getByText, queryByText } = render(<App />);
  const textoAntesDoClique = queryByText("Botão Clicado");
  expect(textoAntesDoClique).toBeNull();

  const botao2 = getByText("Clique aqui");
  fireEvent.click(botao2);

  const textoAposClique = queryByText("Botão Clicado");
  expect(textoAposClique).toBeNull();
});



//Teste Snapshot

test ('Snapshot', async () => {
  const {container} = render(<App/>);

  expect(container).toMatchSnapshot()
})


//Teste de comportamento
test('Teste de comportamento botão', async () => {
  render(<App/>);

  const button2 = screen.getByRole('button', {name: "Clique aqui" })
  expect(button2).toBeInTheDocument()

  fireEvent.click(button2)

  const showingText = screen.queryByText("Botão Clicado")
  expect(showingText).toBeInTheDocument()
});

//Teste que envolve useEffect