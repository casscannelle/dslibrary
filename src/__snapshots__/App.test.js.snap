// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot 1`] = `
<div>
  <div
    class="App"
  >
    <div>
      <header>
        <div
          class="storybook-header"
        >
          <div>
            <svg
              height="32"
              viewBox="0 0 32 32"
              width="32"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                fill="none"
                fill-rule="evenodd"
              >
                <path
                  d="M10 0h12a10 10 0 0110 10v12a10 10 0 01-10 10H10A10 10 0 010 22V10A10 10 0 0110 0z"
                  fill="#FFF"
                />
                <path
                  d="M5.3 10.6l10.4 6v11.1l-10.4-6v-11zm11.4-6.2l9.7 5.5-9.7 5.6V4.4z"
                  fill="#555AB9"
                />
                <path
                  d="M27.2 10.6v11.2l-10.5 6V16.5l10.5-6zM15.7 4.4v11L6 10l9.7-5.5z"
                  fill="#91BAF8"
                />
              </g>
            </svg>
            <h1>
              Acme
            </h1>
          </div>
          <div>
            <button
              class="storybook-button storybook-button--small storybook-button--secondary"
              type="button"
            >
              Log in
            </button>
            <button
              class="storybook-button storybook-button--small storybook-button--primary"
              type="button"
            >
              Sign up
            </button>
          </div>
        </div>
      </header>
      <div
        class="MuiContainer-root MuiContainer-maxWidth1200px css-1ep0w82-MuiContainer-root"
      >
        <h2>
          Página montada para testes
        </h2>
        <p>
          Conteúdo para o teste usando o componente de container.
        </p>
        <button
          class="storybook-button storybook-button--medium storybook-button--primary"
          type="button"
        >
          Clique aqui
        </button>
      </div>
      <div
        class="MuiContainer-root MuiContainer-maxWidth1200px css-1ep0w82-MuiContainer-root"
      >
        <h2>
          Mais um container
        </h2>
        <p>
          Usando o inputfield
        </p>
        <div
          class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          max-width="300"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              id=":r5:"
              placeholder="Nome"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-ihdtdm"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div>
          <button
            class="storybook-button storybook-button--medium storybook-button--secondary"
            type="button"
          >
            Enviar
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
